### PR TITLE
Resolve race condition when joining a node.

### PR DESCRIPTION
**Pull request by cmeiklejohn**
In the event that a joining node happens to trigger an election, which
completes prior to ring convergence, it will never receive the repl
configuration on the source side in time to perform outbound connections
to the sink cluster.

Manually trigger the connection handler code when a ring event occurs,
and the current node is the leader.

This fix, unfortunately, ties riak_repl2_leader to cluster manager,
which is probably undesirable moving forward, but provides a compatible
patch for the upcoming 1.4.4 release.

This is a patch for the 1.4 branch, but should also be ported forward to
2.0.

### CODE REVIEW

**Comment by joedevivo** [comment]
+1 looks good.

**Comment by bsparrow435** [comment]
You could use `riak_core_cluster_mgr:get_is_leader().` instead of riak_repl2_leader if you didn't want them tied together. But this probably has it's own set of tradeoffs.

**Comment by cmeiklejohn** [comment]
Amended commit to remove additional logging.

File: src/riak_core_cluster_mgr.erl [hunk 1]
```diff
@@ -245,6 +251,11 @@ handle_call({get_my_members, MyAddr}, _From, State) ->
 handle_call(leader_node, _From, State) ->
     {reply, State#state.leader_node, State};
 
+handle_call(connect_to_clusters, _From, State) ->
+    lager:info("handling sync connect_to_clusters call"),
```
Comment (bookshelfdave) [inline src/riak_core_cluster_mgr.erl]: maybe a more customer friendly lager msg?

**Comment by bookshelfdave** [comment]
:+1:

File: src/riak_core_cluster_mgr.erl (hunk 1 above)
Comment (cmeiklejohn) [inline src/riak_core_cluster_mgr.erl]: Removed as it's not very necessary.

**Pull request merged**